### PR TITLE
add typing for functions in bps and common tasks

### PR DESCRIPTION
**Pull request by ShujiaoHee**
I added basic type annotations to all functions where typing was missing, but I didn't delve deeply into the structure. So for most functions I didn't provide detailed type information, like the example `List[Type[ProductBased]]`, but I will try to see if there are some improvements. I hope to recieve feedback and suggestions, and would be glad to do the futher improvement. :)

### CODE REVIEW

File: bim2sim/tasks/bps/enrich_material.py [hunk 1]
```diff
@@ -180,7 +183,7 @@ def get_material_template(cls, material_name: str, resumed_names: list,
                                                                material_name)
         return resumed[selected_material]
 
-    def enrich_layer_set(self, invalid_element, resumed, templates):
+    def enrich_layer_set(self, invalid_element: Any, resumed: dict, templates: dict):
```
Comment (ShujiaoHee) [inline bim2sim/tasks/bps/enrich_material.py:186]: I think we will get different types if we run the different cases, so I just set it to "Any".
image

File: bim2sim/tasks/bps/plot_results.py [hunk 2]
```diff
@@ -38,7 +40,7 @@ class PlotBEPSResults(ITask):
     reads = ('df_finals', 'sim_results_path', 'ifc_files', 'elements')
     final = True
 
-    def run(self, df_finals, sim_results_path, ifc_files, elements):
+    def run(self, df_finals: dict, sim_results_path: Path, ifc_files: list, elements: dict):
```
Comment (ShujiaoHee) [inline bim2sim/tasks/bps/plot_results.py:43]: I got the "NoneType" for the parameters "df_finals" and "sim_results_path" when I debug, but acoording the Args above, I just add the `dict` and `Path`. 
image

File: bim2sim/tasks/bps/enrich_material.py
[the remark this answers is not in the document]
Comment (DaJansenGit) [reply bim2sim/tasks/bps/enrich_material.py:1]: Are these errors still existing after the changes you did?

**Comment by DaJansenGit** [comment]
Thanks for the good work! 
I just removed the variable type annotations inside functions (also not recommended [here](https://stackoverflow.com/questions/77653257/are-variable-type-annotations-useful-everywhere)), because I think its redundant and bloating up the code.

File: bim2sim/tasks/bps/enrich_material.py (hunk 1 above)
Comment (DaJansenGit) [reply bim2sim/tasks/bps/enrich_material.py]: Function doesn't exist anymore, so I guess this is obsolete

File: bim2sim/tasks/bps/plot_results.py (hunk 2 above)
Comment (DaJansenGit) [reply bim2sim/tasks/bps/plot_results.py]: this is fine

**Review by DaJansenGit — APPROVED** [review]
From my side this is ready to merge. Please check my answers in your comments if you are fine with this.

**Pull request merged**